jobs/build-charms/charms.py

```diff
@@ -523,14 +542,37 @@ def save(self):
         self.db_json.write_text(json.dumps(dict(self.db)))
         self.store.put_item(Item=dict(self.db))
 
-    def promote_all(self, from_channel="unpublished", to_channels=("edge",)):
+    def promote_all(self, from_channel="beta", to_channels=("edge",)):
         """Promote set of charm artifacts in charmhub."""
+        track = self.db["build_args"].get("track") or "latest"
+        if from_channel.lower() in RISKS:
+            from_channel = f"{track}/{from_channel.lower()}"
+        assert (
+            from_channel != "unpublished"
+        ), "It's unwise to promote unpublished charms."
+        ch_channels = [
+            f"{track}/{chan.lower()}" if (chan.lower() in RISKS) else chan
+            for chan in to_channels
+        ]
+        failed_entities = []
         for charm_map in self.artifacts:
             for charm_name, charm_opts in charm_map.items():
                 if not any(match in self.filter_by_tag for match in charm_opts["tags"]):
                     continue
-                to_channels = self.apply_channel_bounds(charm_name, to_channels)
-                _CharmHub(self).promote(charm_name, from_channel, to_channels)
+                ch_channels = self.apply_channel_bounds(charm_name, ch_channels)
+                try:
+                    _CharmHub(self).promote(charm_name, from_channel, ch_channels)
+                except Exception:
+                    self.echo(traceback.format_exc())
+                    failed_entities.append(charm_name)
```

Comment:
this lets us promote as many charms as possible without stopping just because one of them fails